### DISCUSSION TITLE
Do wizards and scenes work in serverless environments?

### DISCUSSION THREAD

**Discussion by vidrepar** [telegraf/telegraf#1802] [Q&A] [answered]
Hey, I'm just getting into Telegram (telegraf) bots and I have a couple of questions on how things work:

1. Do wizards and scenes work in serverless environments?
2. I'll use webhooks and serverless functions. Where is the data persisted during different (wizard) steps?
3. To what does the `session` refer to? eg. `bot.use(session());`

Thanks for developing this lib. Looks neat.

**Comment by palmerabollo** [comment, marked as answer]
@vidrepar yes, they work as long as you use a session backend. I've successfully used https://www.npmjs.com/package/telegraf-session-s3. There are also plugins for redis, dynamodb, and many other persistence alternatives.

**Reply by vidrepar** [reply]
@palmerabollo Which of these packages have you found the most reliable?

**Comment by MKRhere** [comment]
What @palmerabollo said, but you can just use the recently released official [@telegraf/session](https://www.npmjs.com/package/@telegraf/session) plugin now :)
